sources/api/apiserver/src/server/controller.rs

```diff
@@ -372,6 +401,96 @@ pub(crate) fn set_settings<D: DataStore>(
     let pending = Committed::Pending {
         tx: transaction.into(),
     };
+
+    info!("Writing Metadata to data store");
+    match strength {
+        Strength::Strong => {
+            // Get keys in the request
```

Comment:
The checks against `Live` happening in this function won't work -- we need to handle checks against strength changes at `commit` time. If we do it at `set`, another transaction could come first and change our assumptions, leading to a [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) bug.